data-model/src/test/java/org/pdxfinder/graph/dao/TreatmentProtocolTest.java

```diff
@@ -0,0 +1,295 @@
+        drug1.setTreatment(treatments.get(1));
+        drug2.setTreatment(treatments.get(2));
+
+        componentsList.add(control);
+        componentsList.add(drug1);
+        componentsList.add(drug2);
+
+        Response testResponse = new Response(RESPONSE_STRING, treatmentProtocol);
+
+        treatmentProtocol.setComponents(componentsList);
+        treatmentProtocol.setResponse(testResponse);
+
+        return treatmentProtocol;
+    }
+
+    private void createMockDatasets2() {
+
+        Treatment treatment;
+        treatment = createTreatmentsAndOntologyRelationship();
+        treatmentProtocolList.addAll(createTreatmentComponentsAndProtocols2(treatment));
+    }
+
+    private Treatment createTreatmentsAndOntologyRelationship() {
+
+        Treatment treatment1 = new Treatment(FIRST_TREATMENT_NAME);
+        TreatmentToOntologyRelationship ontologyRelationship1 = new TreatmentToOntologyRelationship();
+        treatment1.setTreatmentToOntologyRelationship(ontologyRelationship1);
+
+        return treatment1;
+    }
+
+    private List<TreatmentProtocol> createTreatmentComponentsAndProtocols2(Treatment treatments){
+
+        TreatmentProtocol treatmentProtocol = new TreatmentProtocol();
+        TreatmentProtocol treatmentProtocol2 = new TreatmentProtocol();
+
+        List<TreatmentComponent> componentsList1 = new ArrayList<>();
+        List<TreatmentComponent> componentsList2 = new ArrayList<>();
+
+        TreatmentComponent control = new TreatmentComponent();
+        TreatmentComponent drug1 = new TreatmentComponent();
+
+        control.setType("Control");
+        control.setTreatment(treatments);
+
+        control.setDose(CONTROL_DOSE_1);
+        control.setDuration(CONTROL_DURATION_1);
+
+        componentsList1.add(control);
+        componentsList2.add(drug1);
+
+        Response testResponse = new Response(RESPONSE_STRING, treatmentProtocol);
+        Response testResponse2 = new Response(RESPONSE_STRING, treatmentProtocol2);
+
+        treatmentProtocol.setComponents(componentsList1);
+        treatmentProtocol2.setComponents(componentsList2);
+
+        treatmentProtocol.setResponse(testResponse);
+        treatmentProtocol2.setResponse(testResponse2);
+
+        return Arrays.asList(treatmentProtocol,treatmentProtocol2);
+    }
+
+    @Test
+    public void Given_getTreatmentString_When_MultipleTCandControlsOn_Then_ReturnNotNullConcatenatedString(){
+
+        String treatmentString;
+
+        treatmentString = treatmentProtocolList
+                .get(0)
+                .getTreatmentString(READ_CONTROLS_COMPONENTS);
+
+        Assert.assertNotNull(treatmentString);
+        Assert.assertEquals(
+                String.format("%s and %s and %s", CONTROL_ONTOLOGY_LABEL, DRUG_ONTOLOGY_LABEL, DRUG_ONTOLOGY_LABEL_2),
+                treatmentString);
+    }
+
+    @Test
+    public void Given_getTreatmentString_WhenTreatmentComponentsWithNoControl_Then_returnConcatenatedStrings(){
```

Comment:
`When` needs a following underscore
```suggestion
    public void Given_getTreatmentString_When_TreatmentComponentsWithNoControl_Then_returnConcatenatedStrings(){
```